phoenix/src/main.rs

```diff
@@ -1,661 +1,200 @@
+
+    // --- SBG Setup ---
+    let mut uart_config = UartConfig::default();
+    uart_config.baudrate = 115200;
+    let usart = Uart::new(
+        p.UART4,
+        p.PA1,
+        p.PA0,
+        Irqs,
+        p.DMA1_CH1,
+        p.DMA1_CH0,
+        uart_config,
+    )
+    .unwrap();
+    let (tx, rx) = usart.split();
+    let ring_rx = rx.into_ring_buffered(unsafe { &mut RX_SBG_BUF });
+    let sbg_pwr = Output::new(p.PD8, Level::High, Speed::Low);
+
+    // // --- Baro SPI Setup ---
+    let mut spi_config = SpiConfig::default();
+    spi_config.frequency = mhz(16);
+    let spi_bus = Spi::new_blocking(p.SPI4, p.PE2, p.PE6, p.PE5, spi_config);
+    info!("SPI4 bus configured.");
+    let baro_cs = Output::new(p.PB8, Level::High, Speed::Low);
+    info!("Barometer CS pin configured.");
+    let baro = Ms5611::new(spi_bus, baro_cs, Delay).unwrap();
+
+    // --- SD Card ---
+    let sd_card = sd::setup_sdmmc_interface(p.SPI1, p.PA5, p.PA7, p.PA6, p.PE9);
+
+    // --- GPS Setup ---
+    let (ring_gps_rx, gps_tx) = sensors::gps::setup_gps(
+        p.PA4, p.PB2, p.UART8, p.PE0, p.PE1, p.DMA1_CH5, p.DMA1_CH6, Irqs,
+    )
+    .await;
+
+    // --- Recovery manager ---
+    let recovery_manager = RecoveryManager::new(
+        p.PD6, p.PD14, p.PC11, p.PD2, p.PD5, p.PD13, p.PC12, p.PD1, p.PA2, p.PB0, p.PA3, p.PC5,
+        p.ADC1, p.PC1,
+    );
+
+    RECOVERY_MANAGER.lock(|cell| {
+        *cell.borrow_mut() = Some(recovery_manager);
+    });
+
+    // --- Camera Triggers ---
+    let cameras = Cameras::new(p.PE14, p.PE12);
+
+    // --- Buzzer 🐝 ---
+    let buzz_out_pin = PwmPin::new_ch1(p.PC6, OutputType::PushPull);
+    let mut pwm = SimplePwm::new(
+        p.TIM3,
+        Some(buzz_out_pin),
+        None,
+        None,
+        None,
+        khz(4),
+        Default::default(),
+    );
+    let mut ch1 = pwm.ch1();
+    info!("Duty Cycle: {}", ch1.max_duty_cycle());
+    ch1.set_duty_cycle(ch1.max_duty_cycle() / 4);
+    ch1.enable();
+    music::play_song(&mut pwm, music::TWINKLE_MELODY, 130).await;
+
+    // --- State Machine ---
+    let state_machine = StateMachine::new(state_machine::Context {});
+
+    // --- Radio ---
+    let (radio_tx, radio_ring_rx) =
+        communication::init_radio(p.UART7, p.PE7, p.PE8, p.DMA2_CH3, p.DMA2_CH5, Irqs);
+
+    // --- Spawning Tasks ---
+    // spawner.must_spawn(led_blinker_task(p.PB14));
+
+    // spawner.must_spawn(uart_dma_reader_task(ring_rx));
+    // spawner.must_spawn(uart_gps_dma_reader_task(ring_gps_rx, gps_tx));
+    // spawner.must_spawn(sbg_parser_task(tx));
+    // spawner.must_spawn(sbg_receiver_task());
+    // spawner.must_spawn(baro_reader_task(baro));
+    // spawner.must_spawn(ai_task());
+    // pass control of the spawner to the state machine
+    // spawner.must_spawn(sm_task(spawner, state_machine));
+    spawner.must_spawn(radio_reader_task(radio_ring_rx));
+    spawner.must_spawn(radio_writer_task(radio_tx));
```

Comment:
Why not uncomment and `#[cfg(feature = "gps")]` `sbg` `baro` etc? Just set the default features.
If you want to have something in the codebase to experiment just add `--features "my-hardcore-temporary-tracing"` etc